### PR TITLE
Add VTEX provider

### PR DESCRIPTION
**Pull request by victormacedo**
Provider for https://vtex.com/
Developer documentation: https://developers.vtex.com/vtex-rest-api/docs/login-integration-guide-webstore-oauth2

### CODE REVIEW

File: src/Vtex/Provider.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+     * Build a complete endpoint concatenating storefront domain from services
+     * configuration and path.
+     *
+     * @param string $path
+     *
+     * @return string
+     */
+    private function appendStorefrontDomain(string $path): string
+    {
+        return $this->getConfig('storefront_domain').$path;
```
Comment (atymic) [inline src/Vtex/Provider.php:80]: Need to add this to config keys

```
public static function additionalConfigKeys()
    {
        return ['storefront_domain'];
    }
``

File: src/Vtex/composer.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    ],
+    "license": "MIT",
+    "authors": [
+        {
+            "name": "Victor Macedo de Moura",
+            "email": "victormacedomoura@gmail.com"
+        }
+    ],
+    "require": {
+        "php": "^7.4 || ^8.0",
```
Comment (atymic) [inline src/Vtex/composer.json:19]:
```suggestion
        "php": "^8.0",
```

**Review by atymic — CHANGES_REQUESTED** [review]

File: src/Vtex/Provider.php [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+namespace SocialiteProviders\Vtex;
+
+use Laravel\Socialite\Two\AbstractProvider;
+use Laravel\Socialite\Two\ProviderInterface;
+use Laravel\Socialite\Two\User;
+
+class Provider extends AbstractProvider implements ProviderInterface
+{
+    /**
```
Comment (lucasmichot) [inline src/Vtex/Provider.php:11]: No `IDENTIFIER` const ?

File: src/Vtex/Provider.php [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+     *
+     * @param string $token
+     *
+     * @return array
+     */
+    protected function getUserByToken($token)
+    {
+        $response = $this->getHttpClient()->get(
+            $this->appendStorefrontDomain('/_v/oauth2/introspect'),
+            ['body' => 'token='.$token]
```
Comment (lucasmichot) [inline src/Vtex/Provider.php:64]: ????

Comment (lucasmichot) [inline src/Vtex/Provider.php:64]: Can you check https://developers.vtex.com/vtex-rest-api/docs/login-integration-guide-webstore-oauth2 ?

File: src/Vtex/Provider.php [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        );
+
+        return json_decode($response->getBody(), true);
```
Comment (lucasmichot) [inline src/Vtex/Provider.php:67]:
```suggestion
        return json_decode((string) $response->getBody(), true);
```

File: src/Vtex/Provider.php (hunk 1 above)
Comment (lucasmichot) [reply src/Vtex/Provider.php:80]: Please throw an exception if `storefront_domain` is not set

**Review by lucasmichot — CHANGES_REQUESTED** [review]

**Comment by atymic** [comment]
ping @victormacedo

**Comment by victormacedo** [comment]
> ping @victormacedo

I'm going to back on this project in a few weeks. If its possible to wait until there I'm going to resolve all changes requested. Thanks

**Comment by atymic** [comment]
cc @victormacedo

**Comment by atymic** [comment]
cc @victormacedo

**Comment by victormacedo** [comment]
I'm waiting a response from VTEX about this feature which is not working properly and the endpoints here are changing by them. I thinks that it's better to close this PR and when its possible I'll open a new one.

**Pull request closed without merging**